transifex/api/README.md

```diff
@@ -0,0 +1,1351 @@
+
+```python
+languages = project.fetch('languages')
+```
+
+### Changing attributes
+
+Lets use what we've learned so far alongside the API documentation to find a
+"string translation _slot_":
+
+```python
+language_dict = {
+    language.code: language
+    for language in project.fetch('languages')
+}
+language = language_dict['el']
+
+resource_dict = {
+    resource.slug: resource
+    for resource in transifex_api.Resource.filter(project=project).all()
+}
+resource = resource_dict['my_resource']
+
+translations = transifex_api.ResourceTranslation.\
+    filter(resource=resource, language=language).\
+    include('resource_string')
+translation = translations[0]
+```
+
+Notes:
+
+1. We don't use the "dict trick" with projects but we do with organizations,
+   resources and languages because there is a `slug` filter for projects in the
+   API but there is not `slug` or `code` filter for the rest (yet). You should
+   consult the API documentation to find out which filters are available
+2. Appending a `.include` to a filter will prefetch a relationship. In the case
+   of ResourceTranslation, this will also fetch the source string information
+   for the "translation slot". Again, you should consult the API documentation
+   to see if including relationships is supported for a given API resource type
+
+In order to save a translation to the server, we use `.save`:
+
+```python
+# We don't have to fetch the resource string because it has been included
+source_string = translation.resource_string.strings['other']
+
+translation.strings = {'other': source_string + " in greeeek!!!"}
+translation.save('strings')
+```
+
+We have to specify which fields we will be sending to the API with `save`'s
+positional arguments.
+
+Because this is a common use-case (setting attributes and immediately saving
+them on the server), there is a shortcut:
+
+```python
+translation.save(strings={'other': source_string + " in greeek!!!"})
+```
+
+### Changing relationships
+
+Lets use projects, teams and project languages as examples:
+
+```
+project = transifex_api.Project.get(...)
+team_1 = project.fetch('team')
+team_2 = transifex_api.Team.filter(...)[0]
+```
+
+If we want to change the project's team from `team_1` to `team_2`, we have 2
+options:
+
+```python
+project.team = team_2
+project.save('team')
+
+# Or
+
+project.save(team=team_2)
+```
+
+This is similar to how we change attributes. The other option is:
+
+```python
+project.change('team', team_2)
+```
+
+This will send a PATCH request to
+[`/projects/XXX/relationships/team`](https://transifex.github.io/openapi/#tag/Projects/paths/~1projects~1{project_id}~1relationships~1team/patch)
+to perform the change. Again, you should consult the API documentation to see
+which relationships can be changed and with which methods (in this case -
+changing a project's team - both methods are available).
+
+The `project -> team` is a "singular relationship". To change a "plural
```

Comment:
```suggestion
The `project -> team` is an "one-to-one relationship". To change an "one-to-many
```

Reply:
A "singular" relationship can either be a one-to-one or a foreign-key relationship. In this case it means that a project belongs to **one** team. It doesn't say whether other projects can or cannot belong to the same team.